app/utils/callback/lava.py

```diff
@@ -70,6 +71,15 @@
     models.BOARD_KEY: "platform.name",
 }
 
+# This is used to detect that an early job step timed out and set the status to
+# UNKNOWN.  It should eventually be reported as an infrastructure error by
+# LAVA, at which point this code could be removed.
+TIMEOUT_RE = re.compile(r"^(.*) timed out after [\d]* seconds$")
+EARLY_TIMEOUT_STEPS = [
+    "http-download",
+    "bootloader-commands",
```

Comment:
The problem is with kernels that fail between the end of the bootloader-commands, and the start of auto-login-action. The ones that (in uboot as an example) output nothing after "Starting kernel". LAVA will report the bootloader-commands step as the failure as it hasn't matched the line which ends that action.

Reply:
OK I need to look at more failed logs.

When the bootloader prompts are set correctly and the bootloader does get stuck, a bootloader-commands timeout should still be treated as UNKNOWN.  I'll see if I can get some other reliable meta-data in the case where this is actually a kernel error.

Reply:
Yes, your problem is that LAVA doesn't differentiate between some of the commands going wrong, and the kernel just not booting. I think it's a failing in LAVA, I'll bring it up at the team meeting tomorrow, we should be able to give a result for this.

Reply:
Adding this to the default set of bootloader prompts would already help a lot I guess (copying from some device dictionary here):
```
{% set boot_message = 'Starting kernel' %}
```

Reply:
Yes for uboot devices adding that is probably all that's required.
GRUB nor iPXE output anything, so my thinking was to log a result once LAVA has sent all of the boot commands and if you've got there successfully, you have to assume anything after that is a kernel error. I'll see what other ideas the team have tomorrow.